src/main/common/strtol.c

```diff
@@ -68,64 +63,53 @@ unsigned long _strto_l(const char * str, char ** endptr, int base, int sflag)
             }
         }
     }
-
     number = 0;
-
     if (((unsigned)(base - 2)) < 35) { /* Legal base. */
         cutoff_digit = ULONG_MAX % base;
         cutoff = ULONG_MAX / base;
         do {
             digit = (        (*str - '0') <= 9)
-                ? /* 0..9 */ (*str - '0')
-                : /* else */ (((0x20 | *str) >= 'a') /* WARNING: assumes ascii. */
-                              ? /* >= A/a */ ((0x20 | *str) - ('a' - 10))
-                              : /* else   */ 40 /* bad value */);
-
+                    ? /* 0..9 */ (*str - '0')
+                    : /* else */ (((0x20 | *str) >= 'a') /* WARNING: assumes ascii. */
+                                  ? /* >= A/a */ ((0x20 | *str) - ('a' - 10))
+                                  : /* else   */ 40 /* bad value */);
             if (digit >= base) {
                 break;
             }
-
             SET_FAIL(++str);
-
             if ((number > cutoff)
-                || ((number == cutoff) && (digit > cutoff_digit))) {
+                    || ((number == cutoff) && (digit > cutoff_digit))) {
                 number = ULONG_MAX;
                 negative &= sflag;
             } else {
                 number = number * base + digit;
             }
         } while (1);
     }
-
 #if _STRTO_ENDPTR
     if (endptr) {
         *endptr = (char *) fail_char;
     }
 #endif
-
     {
         unsigned long tmp = (negative
-                             ? ((unsigned long)(-(1+LONG_MIN)))+1
+                             ? ((unsigned long)(-(1 + LONG_MIN))) + 1
                              : LONG_MAX);
         if (sflag && (number > tmp)) {
             number = tmp;
         }
     }
-
     return negative ? (unsigned long)(-((long)number)) : number;
 }
 
-long strtol(const char * str, char ** endptr, int base)
-{
+long strtol(const char * str, char ** endptr, int base) {
     return _strto_l(str, endptr, base, 1);
 }
 
-unsigned long strtoul(const char * str, char ** endptr, int base)
-{
+unsigned long strtoul(const char * str, char ** endptr, int base) {
     return _strto_l(str, endptr, base, 0);
 }
 
-int atoi(const char *str)
-{
+int atoi(const char *str) {
```

Comment:
Codacy found an issue: [Unless checked, the resulting number can exceed the expected range (CWE-190).  If source untrusted, check both minimum and maximum, even if the input had no minus sign (large numbers can roll over into negative number; consider saving to an unsigned value if that is intended). ](https://app.codacy.com/gh/emuflight/EmuFlight/pullRequest?prid=6307959)